dynamicprogramming/fibonacci.go

```diff
@@ -0,0 +1,21 @@
+package dynamicprogramming
+
+// Nth fibonacci Number
+// https://www.geeksforgeeks.org/program-for-nth-fibonacci-number/
+
+func fib(n int) int {
+	/* Declare an array to store Fibonacci numbers. */
+	f := make([]int, n+2) // 1 extra to handle case, n = 0
```

Comment:
- This extra space is useful when the requirement is to return the first n fibonacci numbers.
- But since this program only returns the nth fibonacci number, we don't have to save all the previous numbers. We only need to keep track of the two previous numbers in the series.